### DISCUSSION TITLE
Custom Auth (MagicLabs: Magic Link) with Supabase

### DISCUSSION THREAD

**Discussion by benkimbuilds** [supabase/supabase#10499] [Questions]
I'm trying to integrate Magic Link with Supabase.

Right now, the user signs up/logs in through Magic Link (https://magic.link/). After they successfully log in, I would like to create a session or log the user in through Supabase so that they have RLS authorization.

I mint my own JWT token and use it in the Supabase client initialization.
<img width="821" alt="image" src="https://user-images.githubusercontent.com/32351642/203654380-f4439f2f-fdc1-4794-8e9d-7751ddebb4a6.png">

This is what my next.js api endpoint for logging in looks like:
<img width="836" alt="image" src="https://user-images.githubusercontent.com/32351642/203654485-e9dd8836-1a05-4390-bf12-7c4372ccca53.png">

I created a few debug consoles to inspect the state within supabase:
<img width="859" alt="image" src="https://user-images.githubusercontent.com/32351642/203654541-837829d1-c593-490d-b3c0-590677775046.png">

Any session related query returns an error:
<img width="628" alt="image" src="https://user-images.githubusercontent.com/32351642/203654593-d7ef3dd2-4b62-46ba-8242-13b94f5a6dcb.png">

but my `getUser()` query works prefectly fine.

There's something not clicking with JWT. I wish that Supabase would just make it easy for us and create a `signInWithToken` function.

Am I creating my access token incorrectly?

**Comment by GaryAustin1** [comment]
One thing you might need to check (really not digging thru all of your code), but if this is running on a server you need to initialize the client with persistSession:false option.  Without that, if there is no local storage, the session will get cleared in many cases.

**Comment by minyan-gu** [comment]
I also want to implement sth similar with custom auth capability (not email auth, not phone auth etc..), so I think the request to have a generic function to `signInWithToken` to abstract away the underlying logics would be quite helpful to developers.

Also, I want to confirm the supabase-js `setSession()` behaviour in V2, is it now expecting the user has to be pre-existed in the `auth.user` table before invoking the setSession method? Seems now I got errors after attempting to migrate over to V2.

Just give some background in V1, I only populate users in my own user table and generate JWT myself with custom payload before passing it to the `setSession()` method, everything works fine in V1, Can I do similar things in V2 by not involving the `auth.user` table at all?

**Reply by benkimbuilds** [reply]
I agree, lots of people have requested a `signInWithToken` function.

**Comment by minyan-gu** [comment]
Hi - it's been a long time after releasing v2 now, please can someone from Supabase team provide any comments / guidance here? 

thanks so much in advance.
